neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,298 @@
+            engine.Service.Snapshot.Storages.GetAndChange(CreateAccountKey(account)).Value = state.ToByteArray();
+        }
+
+        private BigInteger CalculateBonus(ApplicationEngine engine, BigInteger value, uint start, uint end)
+        {
+            if (value.IsZero || start >= end) return BigInteger.Zero;
+            if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
+            uint amount = 0;
+            uint ustart = start / Blockchain.DecrementInterval;
+            if (ustart < Blockchain.GenerationAmount.Length)
+            {
+                uint istart = start % Blockchain.DecrementInterval;
+                uint uend = end / Blockchain.DecrementInterval;
+                uint iend = end % Blockchain.DecrementInterval;
+                if (uend >= Blockchain.GenerationAmount.Length)
+                {
+                    uend = (uint)Blockchain.GenerationAmount.Length;
+                    iend = 0;
+                }
+                if (iend == 0)
+                {
+                    uend--;
+                    iend = Blockchain.DecrementInterval;
+                }
+                while (ustart < uend)
+                {
+                    amount += (Blockchain.DecrementInterval - istart) * Blockchain.GenerationAmount[ustart];
+                    ustart++;
+                    istart = 0;
+                }
+                amount += (iend - istart) * Blockchain.GenerationAmount[ustart];
+            }
+            amount += (uint)(engine.Service.Snapshot.GetSysFeeAmount(end - 1) - (start == 0 ? 0 : engine.Service.Snapshot.GetSysFeeAmount(start - 1)));
+            return value * amount * GAS.Factor / TotalAmount;
+        }
+
+        private bool Initialize(ApplicationEngine engine, UInt160 account)
+        {
+            if (engine.Service.Trigger != TriggerType.Application) throw new InvalidOperationException();
+            StorageKey key = CreateStorageKey(Prefix_Initialized);
+            if (engine.Service.Snapshot.Storages.TryGet(key) != null) return false;
+            engine.Service.Snapshot.Storages.Add(key, new StorageItem
+            {
+                Value = new byte[] { 1 },
+                IsConstant = true
+            });
+            MintTokens(engine, account, TotalAmount);
+            foreach (ECPoint pubkey in Blockchain.StandbyValidators)
+                RegisterValidator(engine, pubkey.EncodePoint(true));
+            return true;
+        }
+
+        private BigInteger UnclaimedGas(ApplicationEngine engine, UInt160 account, uint end)
+        {
+            StorageItem storage = engine.Service.Snapshot.Storages.TryGet(CreateAccountKey(account));
+            if (storage is null) return BigInteger.Zero;
+            AccountState state = new AccountState(storage.Value);
+            return CalculateBonus(engine, state.Balance, state.BalanceHeight, end);
+        }
+
+        private bool RegisterValidator(ApplicationEngine engine, byte[] pubkey)
```

Comment:
Don't you think a pubkey type apply here? ECPoint perhaps could be used instead of `byte[]`. This avoids the next unexpected failure.

Reply:
But `ECPoint` is not used in `RegisterValidator` at all.